train_seq2seq.py

```diff
@@ -1,217 +0,0 @@
-from __future__ import print_function
-import sys
-import os
-import collections
-import traceback
-import torch
-import torch.optim as optim
-import torch.nn as nn
-import numpy as np
-from sklearn.metrics import precision_score, accuracy_score, recall_score, mean_squared_error
-from model import Def2VecModel
-from torch.autograd import Variable
-import torchtext.vocab as vocab
-from tensorboardX import SummaryWriter
-from pytorch_monitor import monitor_module, init_experiment
-from loader import *
-import torch.nn.init as init
-from tqdm import tqdm
-from time import time
-from seq2seq import EncoderRNN, DecoderRNN
-
-
-DEBUG_LOG = False
-
-CONFIG = dict(
-    # meta data
-    title="def2vec",
-    description="Translating definitions to word vectors",
-    run_name='full_run_big', 
-    run_comment='def_concat', # gets appended to run_name as RUN_NAME-RUN_COMMENT
-    log_dir='outputs/def2vec/logs',
-    vocab_dim = 100,
-    vocab_source = '6B',
-    load_path = None,
-    # hyperparams
-    random_seed=42,
-    learning_rate=.0001,
-    max_epochs=15,
-    batch_size=64,
-    n_hidden=250,
-    # logging params
-    print_freq=1,
-    write_embed_freq=100,
-    eval_freq = 1000,
-    save_path="model_weights.torch",
-    embedding_log_size = 10000,
-    # data loading params
-    num_workers = 8,
-    packing=True,
-    shuffle=True,
-    # model configuration [for ablation/hyperparam experiments]
-    weight_init="xavier",
-    input_method=INPUT_METHOD_ALL_CONCAT,
-    use_bidirection=True,
-    use_attention=True,
-    cell_type='GRU',
-    #use_batchnorm=True,
-    hidden_size=150,
-    embed_size=100,
-    dropout=0.1,
-    weight_decay=0.0,
-)
-TRAIN_FILE = 'data/glove/train_glove.%s.%sd.txt'%(CONFIG['vocab_source'],CONFIG['vocab_dim'])
-VAL_FILE = 'data/glove/val_glove.%s.%sd.txt'%(CONFIG['vocab_source'],CONFIG['vocab_dim'])
-
-def weights_init(m):
-    """
-    Initialize according to Xavier initialization or default initialization.
-    """
-    if CONFIG['weight_init'] == 'xavier':
-        if type(m) in [nn.Linear]:
-            nn.init.xavier_normal(m.weight.data)
-        elif type(m) in [nn.LSTM, nn.RNN, nn.GRU]:
-            nn.init.xavier_normal(m.weight_hh_l0)
-            nn.init.xavier_normal(m.weight_ih_l0)
-
-
-if __name__ == "__main__":
-
-    vocab = vocab.GloVe(name=CONFIG['vocab_source'], dim=CONFIG['vocab_dim'])
-    vocab_size = len(vocab.stoi)
-    use_gpu = torch.cuda.is_available()
-    print("Using GPU:", use_gpu)
-
-    encoder = EncoderRNN(vocab_size, )
-    model = Seq2Seq(encoder, decoder)
```

Comment:
can you add this bit to train.py? We still need a means by which to train the seq2seq models, I don't think there is one if you just remove both files?

Reply:
Hi @andreykurenkov 
Seems to me the Seq2seq part is already included in train.py. 

What I can see in code is that there are 3 different models:
1, Def2VecModel (in model.py): this should be the original def2vec model;
2, Seq2SeqModel (also in model.py): this looks like to me like the 'SAE' with Attention;
3, BaselineModel (in baseline.py): looks like this is the 'BOW' autoencoder.
The file 'train_seq2seq.py' only appeared in one pull request and is never finished, looks like an early draft but obsolete ever since:
[train_seq2seq.py](https://github.com/andreykurenkov/HybridVec/commit/9f943412f9468262ca54ef122e33923e52ea3cb2#diff-e1792cbdc031f8b6f5da6d78d5d75922)

Please confirm for me, thanks!